src/app/test/test-folder-tree/test-folder-tree.component.ts

```diff
@@ -1,86 +1,97 @@
+    highlightOpenFolders: false,
+    autoSelectCondition: (item: CreateTreeItem) => this.autoSelectItem(item),
+  };
+  originalItems?: CreateTreeItem[];
 
-  constructor() {}
-
-  ngAfterViewInit(): void {
-    this.treeReference.createComponent({ source: [this.baseFolder], height: '90%', width: '100%', allowDrag: false });
-  }
-
-  updateFolderTree(reports: any, addedFolderPath: any) {
-    this.resetFolders();
-    reports.forEach((report: any) => {
-      this.addFolder(report.path);
-    });
-    this.treeReference.clear();
-    this.treeReference.addTo(this.baseFolder, null);
-    this.selectNewFolder(addedFolderPath);
+  autoSelectItem(item: CreateTreeItem): boolean {
+    return item.name === this.rootFolder.name;
   }
 
-  resetFolders() {
-    this.baseFolder = {
-      label: 'Reports',
-      value: '',
-      items: [],
-      expanded: true,
-      selected: true,
-    };
+  selectItem(value: string) {
+    value = this.removeSlashesFromPathEnds(value);
+    const path = value.length > 0 ? `Reports/${value}` : 'Reports';
+    this.tree.selectItem(path);
   }
 
-  addFolder(path: string): void {
-    if (path) {
-      let folderNames = path.startsWith('/') ? path.slice(1).split('/') : path.split('/');
-      folderNames.pop();
-      this.recursivelyAddFolders(folderNames, this.baseFolder.items, '/');
+  removeSlashesFromPathEnds(path: string) {
+    if (path.startsWith('/')) {
+      path = path.slice(1);
     }
+    if (path.endsWith('/')) {
+      path = path.slice(0, -1);
+    }
+    return path;
   }
 
-  recursivelyAddFolders(folderNames: string[], currentFolders: any[], previousFilter: string): void {
-    if (folderNames.length > 0) {
-      let name = folderNames.shift()!;
-      let currentFolder = currentFolders.find((folder) => folder.label === name);
-
-      if (!currentFolder) {
-        currentFolder = this.createNewFolder(name, previousFilter);
-        currentFolders.push(currentFolder);
-      }
-
-      this.latestAddedFolder = currentFolder;
-      this.recursivelyAddFolders(folderNames, currentFolder.items, currentFolder.value);
+  setData(data: CreateTreeItem[]) {
+    this.resetTree();
+    this.originalItems = data;
+    const tempItems = this.convertToFolders(data);
+    let rootFolder = Object.assign({}, this.rootFolder);
+    if (tempItems && tempItems.length > 0) {
+      rootFolder.children = tempItems;
     }
+    this.tree.addItem(rootFolder);
   }
 
-  selectNewFolder(addedFolderPath: string) {
-    if (addedFolderPath) {
-      let item = this.treeReference.getItems().find((item: any) => addedFolderPath === item.value);
-      if (item) {
-        this.treeReference.selectItem(item);
-        this.changeFolderEvent.next(item.value);
+  convertToFolders(items: CreateTreeItem[]) {
+    const tempItems: CreateTreeItem[] = [];
+    for (let item of items) {
+      if (item.path) {
+        try {
+          tempItems.push(this.createFolderFromPath(item.path));
+        } catch {}
```

Comment:
Is an error thrown here? Seems like there is always an array so push should never fail.
Also an empty catch okay in some cases but in most cases you at least want to log something or do a different action, now its even more unclear what the error could be about

Reply:
Shouldnt the try catch then cover the whole function body where the catch returns an empty array? Now you're not handling the error at all and if only one of the items has a wrong path it'll not be obvious that the error has actually occurred which then causes confusion maybe you but definitely anyone else working on this project

Reply:
You can return an empty array in the catch, also with the current code you don't even guarantee that you get an empty array back since other items can have valid paths and thus can the array be filled even with the error having been thrown